lib/writer/block-write-stream.js

```diff
@@ -134,10 +164,24 @@ class BlockWriteStream extends stream.Writable {
     }
 
     fs.write(this.fd, chunk, 0, chunk.length, chunk.position, (error, bytesWritten) => {
-      this.bytesWritten += bytesWritten
-      this.delta += bytesWritten
-      this.blocksWritten += 1
-      this.position += bytesWritten
+      if (!error) {
+        this.bytesWritten += bytesWritten
+        this.delta += bytesWritten
+        this.blocksWritten += 1
+        this.position += bytesWritten
+        this.retries = 0
+        next(error)
+        return
+      }
+
+      const isTransient = BlockWriteStream.isTransientError(error)
+
+      if (isTransient && (this.retries < this.maxRetries)) {
+        this.retries += 1
```

Comment:
Perhaps I'm mis-remembering, or perhaps I'm getting confused with something else, but ISTR that when @jviotti added retry-handling somewhere, he actually put in a small delay between getting the error and retrying the operation, rather than retrying immediately? Would/could that be beneficial here?

Reply:
Aha, found it :smiley: 
https://github.com/resin-io-modules/etcher-image-write/blob/master/lib/filesystem.js#L92

It actually increases the delay as the number of retries increases. And it also does up to 10 retries, rather than the up to 5 that you do here. I'll let you and @jviotti argue about the pros and cons of the different approaches... :wink:

Reply:
Any thoughts @jhermsmeier re: adding a delay between retry attempts?

Reply:
Hmm yep

Reply:
There'll be a delay of a few milliseconds between the retries by default, due to the I/O and it running in the next loop tick; I don't think we need to artificially delay it even more – @jviotti thoughts?